testapp/exam/code_server.py

```diff
@@ -461,29 +109,29 @@ def run_cplus_code(self, answer, test_code, in_dir=None):
         -------
 
         A tuple: (success, error message).
-
         """
-        if in_dir is not None and isdir(in_dir):
-            os.chdir(in_dir)
-
-        # The file extension must be .cpp
-        submit_f = open('submitstd.cpp', 'w')
-        submit_f.write(answer.lstrip())
-        submit_f.close()
-        submit_path = abspath(submit_f.name)
-
-        ref_path = test_code.strip()
-        if not ref_path.startswith('/'):
-            ref_path = join(MY_DIR, ref_path)
+        success = False
+        prev_handler = None
+
+        methods = {"python": 'evaluate_python_code',
```

Comment:
This is not good.  It means we'll have to change this method each time a bloke wants to support a new language.  Just use a subclass where the subclass implements the appropriate `evaluate` method.  You can have perhaps a classmethod called "register" which can be called by subclasses to register their new language implementations.